packages/mui-base/src/Slider/SliderControl/useSliderControl.ts

```diff
@@ -0,0 +1,263 @@
+    // @ts-ignore buttons doesn't not exists on touch event
+    if (nativeEvent.type === 'pointermove' && nativeEvent.buttons === 0) {
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      handleTouchEnd(nativeEvent);
+      return;
+    }
+
+    const { newValue, activeIndex } = getFingerNewValue({
+      finger,
+      move: true,
+      offset: offsetRef.current,
+    });
+
+    focusThumb({ sliderRef: controlRef, activeIndex, setActive });
+
+    if (validateMinimumDistance(newValue, step, minStepsBetweenValues)) {
+      setValueState(newValue);
+
+      if (!dragging && moveCount.current > INTENTIONAL_DRAG_COUNT_THRESHOLD) {
+        setDragging(true);
+      }
+
+      if (handleValueChange && !areValuesEqual(newValue)) {
+        handleValueChange(newValue, activeIndex, nativeEvent);
+      }
+    }
+  });
+
+  const handleTouchEnd = useEventCallback((nativeEvent: TouchEvent | PointerEvent) => {
+    const finger = trackFinger(nativeEvent, touchId);
+    setDragging(false);
+
+    if (!finger) {
+      return;
+    }
+
+    const { newValue } = getFingerNewValue({
+      finger,
+      move: true,
+    });
+
+    setActive(-1);
+    if (nativeEvent.type === 'touchend') {
+      setOpen(-1);
+    }
+
+    if (onValueCommitted) {
+      onValueCommitted(newValue, nativeEvent);
+    }
+
+    touchId.current = undefined;
+
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    stopListening();
+  });
+
+  const handleTouchStart = useEventCallback((nativeEvent: TouchEvent) => {
+    if (disabled) {
+      return;
+    }
+
+    const touch = nativeEvent.changedTouches[0];
+
+    if (touch != null) {
+      touchId.current = touch.identifier;
+    }
+
+    const finger = trackFinger(nativeEvent, touchId);
+
+    if (finger !== false) {
+      const { newValue, activeIndex } = getFingerNewValue({
+        finger,
+      });
+      focusThumb({ sliderRef: controlRef, activeIndex, setActive });
+
+      setValueState(newValue);
+
+      if (handleValueChange && !areValuesEqual(newValue)) {
+        handleValueChange(newValue, activeIndex, nativeEvent);
+      }
+    }
+
+    moveCount.current = 0;
+    const doc = ownerDocument(controlRef.current);
+    doc.addEventListener('touchmove', handleTouchMove, { passive: true });
+    doc.addEventListener('touchend', handleTouchEnd, { passive: true });
+  });
+
+  const stopListening = React.useCallback(() => {
+    offsetRef.current = 0;
+    const doc = ownerDocument(controlRef.current);
+    doc.removeEventListener('pointermove', handleTouchMove);
+    doc.removeEventListener('pointerup', handleTouchEnd);
+    doc.removeEventListener('touchmove', handleTouchMove);
+    doc.removeEventListener('touchend', handleTouchEnd);
+  }, [handleTouchEnd, handleTouchMove]);
+
+  React.useEffect(() => {
+    const { current: control } = controlRef;
+    control!.addEventListener('touchstart', handleTouchStart, {
```

Comment:
Let's try to avoid `!` and use early-returns instead

Reply:
Updated!